### DISCUSSION TITLE
Project with path ':webrtc-kmp' could not be found in project ':shared'.

### DISCUSSION THREAD

**Discussion by sweiss435** [shepeliev/webrtc-kmp#92] [General]
Hello, I spent the weekend trying to integrate with your latest.  I renamed the cartfile to remove the Carthage stuff, but if I use the specs in the readme with export(project(":webrtc-kmp")) I just get Project with path ':webrtc-kmp' could not be found in project ':shared'.

That seems to work with your sample project because you have a module named webrtc-kmp, but for those of us trying to pull it into a standalone project, I don't see how this makes sense.

If I comment that out I can run the android app, but when I go to run the iosapp after pod deintegrate/install I get the following in Xcode
Execution failed for task ':shared:podGenIOS'.
> Executing of 'pod install' failed with code 1 and message: 
      [33mWARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
      Consider adding the following to ~/.profile:
      export LANG=en_US.UTF-8

I've tried everything out there editing my .zshrc etc, and this is just a warning, no error seems to be mentioned, but I can't run the iOS app

**Comment by shepeliev** [comment]
Hello. Oh, I've copy paste config from sample project, but not changed `export` arg. Definitely, for standalone project it should be as following:

```
export("com.shepeliev:webrtc-kmp:$webRtcKmpVersion")
```

Thanks for highlighting. I'll fix readme.

**Comment by sweiss435** [comment]
Thanks, tried that, still seem to get a mystery error when trying to run it from Xcode.  Shows me some warnings, no error, but says FAILURE: Build failed with an exception.

I created a brand new stripped down KMM application in Android studio and the only thing I changed was adding your webrtc-kmp stuff in as instructed.  Project is attached.  The basic project seems to fail in the same manner as my project that had worked before when this was on cocoa pods and recently with Carthage.  

Can you try it out and see what's going wrong?

[WebRtcTest.zip](https://github.com/shepeliev/webrtc-kmp/files/12235135/WebRtcTest.zip)

**Reply by shepeliev** [reply]
Yes, I'll look at this soon. Thanks for sample project.

**Reply by shepeliev** [reply]
So I've checked it. Looks like the core reason is https://github.com/CocoaPods/CocoaPods/issues/10939

AFAIK, xCode runs in own environment and its inherited by gradlew script. It should be fixed in Kotlin 1.9.20 if this [issue](https://youtrack.jetbrains.com/issue/KT-59522/Set-the-required-environment-for-cocoapods-invocations) doesn't lie 😀.

For now the following workaround works (at least I could build the test project on my Intel mac):

```bash
launchctl setenv LANG en_US.UTF-8
``` 

Just in case, don't forget too add `api("com.shepeliev:webrtc-kmp:0.114.4")` in common dependencies in shared module of the test project.


<img width="599" alt="Знімок екрана 2023-08-02 о 19 00 14" src="https://github.com/shepeliev/webrtc-kmp/assets/2721878/7a9cb92a-25c6-4c26-a7c0-5a1273d517e1">
